### PR TITLE
test(integration tests): sapling multiple states test

### PR DESCRIPTION
**Pull request by roxaneletourneau**
An integration test to interact with a contract having 2 sapling pools

re #1825

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [ ] Your code builds cleanly without any errors or warnings
- [ ] You have added unit tests
- [ ] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+CONFIGS().forEach(({ lib, rpc, setup }) => {
+  const Tezos = lib;
+  let saplingContract: ContractAbstraction<ContractProvider>;
+  let saplingStateIdLeft: string;
+  let saplingStateIdRight: string;
+  let aliceInmemorySpendingKey: InMemorySpendingKey;
+  let alicePaymentAddress: string;
+  const memoSize = 8;
+
+  describe(`Test interaction with sapling contract having multiple sapling states using: ${rpc}`, () => {
```
Comment (dsawali) [inline integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts:18]: What do you think about a shorter describe block name? Something like 
'Sapling contract interaction - multiple sapling states'

**Review by dsawali — CHANGES_REQUESTED** [review]

File: integration-tests/sapling-transactions-contract-with-single-state.spec.ts [hunk 2]
```diff
@@ -32,7 +32,7 @@ CONFIGS().forEach(({ lib, rpc, setup }) => {
       const mnemonic: string = bip39.generateMnemonic();
       bobInmemorySpendingKey = await InMemorySpendingKey.fromMnemonic(mnemonic);
 
-      // Instanciate an InMemorySpendingKey from a spending key for Alice
+      // Instantiate an InMemorySpendingKey from a spending key for Alice
```
Comment (zainen) [inline integration-tests/sapling-transactions-contract-with-single-state.spec.ts:35]: i spell this wrong all the time too 😓

**Review by zainen — COMMENTED** [review]
LGTM

File: integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts (hunk 1 above)
Comment (roxaneletourneau) [reply integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts:18]: Yeah maybe we should have a discussion about that. I try to stick to the format Michael is using for changing all tests description.

File: integration-tests/sapling-transactions-contract-with-single-state.spec.ts (hunk 2 above)
Comment (roxaneletourneau) [reply integration-tests/sapling-transactions-contract-with-single-state.spec.ts:35]: I always do this mistake :P

File: integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts (hunk 1 above)
Comment (dsawali) [reply integration-tests/sapling-transactions-contract-with-multiple-sapling-states.spec.ts:18]: yeah that sounds good

**Review by dsawali — APPROVED** [review]

**Pull request merged**